community/ndp/transport-http/src/test/java/org/neo4j/ndp/transport/http/util/HTTP.java

```diff
@@ -0,0 +1,363 @@
+/**
+ * Copyright (c) 2002-2015 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.ndp.transport.http.util;
+
+import com.sun.jersey.api.client.Client;
+import com.sun.jersey.api.client.ClientRequest;
+import com.sun.jersey.api.client.ClientResponse;
+import org.apache.commons.io.Charsets;
+import org.codehaus.jackson.JsonNode;
+
+import java.net.URI;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.ws.rs.core.MediaType;
+
+import org.neo4j.server.rest.domain.JsonHelper;
+import org.neo4j.server.rest.domain.JsonParseException;
+import org.neo4j.ndp.messaging.v1.util.BytePrinter;
+
+import static java.util.Collections.unmodifiableMap;
+import static org.hamcrest.Matchers.allOf;
+import static org.hamcrest.Matchers.anyOf;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.not;
+import static org.junit.Assert.assertThat;
+import static org.neo4j.helpers.collection.IteratorUtil.singleOrNull;
+import static org.neo4j.helpers.collection.MapUtil.stringMap;
+
+/**
+ * A tool for performing REST HTTP requests
+ */
+public class HTTP
```

Comment:
I think I should expand this a bit further: I don't want to use the server HTTP client here & now, because it would require adding a dependency on the server, which will become a circular dependency if we pull this into the Neo4j Server. The right thing to do would be to have something like a test-util module for the HTTP stuff, but it seems that's a bit of overhead to add since the plan is to remove this altogether in the next two weeks or so.